roc.go

```diff
@@ -0,0 +1,154 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package stat
+
+import "sort"
+
+// SortWeightedBool rearranges the data in y along with their
+// corresponding boolean weights c so that the y data are sorted.
+// The data is sorted in place.
+// c may be nil, but if c is non-nil then it must have the same
+// length as y.
+func SortWeightedBool(y []float64, c []bool) {
+	if c == nil {
+		sort.Float64s(y)
+		return
+	}
+	if len(y) != len(c) {
+		panic("stat: slice length mismatch")
+	}
+	sort.Sort(byY{
+		y: y,
+		c: c,
+	})
+}
+
+type byY struct {
+	y []float64
+	c []bool
+}
+
+func (a byY) Len() int { return len(a.y) }
```

Comment:
The Float64Slice implementation in sort goes like this:

```
// Float64Slice attaches the methods of Interface to []float64, sorting in increasing order.
type Float64Slice []float64

func (p Float64Slice) Len() int           { return len(p) }
func (p Float64Slice) Less(i, j int) bool { return p[i] < p[j] || isNaN(p[i]) && !isNaN(p[j]) }
func (p Float64Slice) Swap(i, j int)      { p[i], p[j] = p[j], p[i] }
```

Should we adopt this style? I'm not fussed either way but I do think we should be consistent.